test/components/ChartForm_test.js

```diff
@@ -0,0 +1,383 @@
+        expect(form.text()).toMatch('Dates must be MM/DD/YYYY format.');
+        expect(form.find(selector.saveButton).prop('disabled')).toBe(true);
+      });
+
+      it('validates chart end date formats', () => {
+        // enter an invalid date
+        form.find('input[name=chart_end_date]').val('02/30/2017').change();
+
+        expect(form.find('.has-error').length).toEqual(1);
+        expect(form.text()).toMatch('Dates must be MM/DD/YYYY format.');
+        expect(form.find(selector.saveButton).prop('disabled')).toBe(true);
+      });
+
+      it('disables saving when target date validation fails', () => {
+        // make a change to enable save button
+        form.find(selector.titleField).val('Test Title').change();
+        expect(form.find(selector.saveButton).prop('disabled')).toBe(false);
+
+        // delete the start date, which is required when target is present
+        form.find(selector.startDateField).val('').change();
+
+        // errors should disable save button
+        expect(form.find('.has-error').length).toEqual(1);
+        expect(form.find(selector.saveButton).prop('disabled')).toBe(true);
+      });
+
+      it('clears errors on cancel', () => {
+        // delete the title
+        form.find(selector.titleField).val('').change();
+
+        expect(form.find('.has-error').length).toEqual(1);
+        expect(form.find(selector.saveButton).prop('disabled')).toBe(true);
+
+        // click the cancel button
+        form.find(selector.cancelButton).click();
+
+        expect(form.find('.has-error').length).toEqual(0);
+      });
+    });
+
+    it('uses jQuery plugin for date inputs', () => {
+      let spyObj = {
+        datepicker() {}
+      };
+
+      spyOn(spyObj, 'datepicker');
+
+      // eslint-disable-next-line no-undef
+      $.fn.datepicker = spyObj.datepicker;
+
+      shallowMount(ChartForm, {
+        propsData: {
+          chartDef: exampleChart,
+          metadata: exampleMetadata
+        }
+      });
+
+      expect(spyObj.datepicker).toHaveBeenCalled();
+    });
+  });
+
+  describe('with longitudinal project', () => {
+    let wrapper;
+
+    beforeEach(() => {
+      wrapper = shallowMount(ChartForm, {
+        propsData: {
+          chartDef: exampleLongitudinalChart,
+          metadata: exampleLongitudinalMetadata
+        }
+      });
+    });
+
+    it('displays the event dropdown for longitudinal projects', () => {
+      expect(wrapper.findAll(selector.dateFieldEventSelect).length).toEqual(1);
+      expect(wrapper.findAll(selector.groupFieldEventSelect).length).toEqual(1);
+    });
+
+    it('creates the expected date field options for the selected event', () => {
+      const dateFieldOptions = wrapper.findAll(`${selector.dateFieldSelect} > option`);
+      expect(dateFieldOptions.length).toEqual(2); // 1 options plus prompt
+      expect(dateFieldOptions.at(1).text().trim()).toEqual('survey_date (Survey date)'); // sorted by label
+    });
+
```

Comment:
I had a terrible time getting this test and the corresponding test for the grouping field event to work. The code worked fine when I interacted with it in a browser, but the tests were throwing exceptions. Nothing I tried yesterday seemed to work. In desperation last night, I moved code to copy the `chartDef` prop from the `mounted()` hook to the `data()` hook, which fixed these tests. I'm not certain what the root cause was, but it may have been caused by computed properties depending on data that wasn't present until the component mounted.